stl/inc/xstring

```diff
@@ -2744,33 +2798,34 @@ private:
         _CSTD memcpy(_My_data_mem, _Right_data_mem, _Memcpy_val_size);
     }
 
-    void _Take_contents(basic_string& _Right, true_type) noexcept {
-        // assign by stealing _Right's buffer, memcpy optimization
-        // pre: this != &_Right
-        // pre: allocator propagation (POCMA) from _Right, if necessary, is complete
-        // pre: *this owns no memory, iterators orphaned (note:
-        // _Buf/_Ptr/_Mysize/_Myres may be garbage init)
-#if _ITERATOR_DEBUG_LEVEL != 0
-        if (_Right._Mypair._Myval2._Large_string_engaged()) {
-            // take ownership of _Right's iterators along with its buffer
-            _Swap_proxy_and_iterators(_Right);
-        } else {
-            _Right._Mypair._Myval2._Orphan_all();
-        }
-#endif // _ITERATOR_DEBUG_LEVEL != 0
-
-        _Memcpy_val_from(_Right);
-        _Right._Tidy_init();
-    }
-
-    void _Take_contents(basic_string& _Right, false_type) noexcept {
-        // assign by stealing _Right's buffer, general case
+    _CONSTEXPR20_CONTAINER void _Take_contents(basic_string& _Right) noexcept {
+        // assign by stealing _Right's buffer
         // pre: this != &_Right
         // pre: allocator propagation (POCMA) from _Right, if necessary, is complete
         // pre: *this owns no memory, iterators orphaned
         // (note: _Buf/_Ptr/_Mysize/_Myres may be garbage init)
         auto& _My_data    = _Mypair._Myval2;
         auto& _Right_data = _Right._Mypair._Myval2;
+#ifdef __cpp_lib_constexpr_string
+        if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_constexpr_string
+        {

```

Comment:
Style question: Should we put this branch inside the `if constexpr`? Would that improve throughput?